test/WebJobs.Script.Tests/FileProvisioning/PowerShellFileProvisionerTests.cs

```diff
@@ -2,41 +2,135 @@
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
 using System;
+using System.Collections.Generic;
 using System.IO;
+using System.Linq;
 using System.Threading.Tasks;
 using Microsoft.Azure.WebJobs.Script.FileProvisioning.PowerShell;
+using Microsoft.EntityFrameworkCore.Internal;
+using Microsoft.Extensions.Logging;
 using Xunit;
 
 namespace Microsoft.Azure.WebJobs.Script.Tests.FileAugmentation
 {
     public class PowerShellFileProvisionerTests
     {
+        private const string ProfilePs1FileName = "profile.ps1";
+        private const string RequirementsPsd1FileName = "requirements.psd1";
+
         private readonly string _scriptRootPath;
+        private TestLogger _logger;
 
         public PowerShellFileProvisionerTests()
         {
+            _logger = new TestLogger("PowerShellFileProvisionerTests");
             _scriptRootPath = Path.GetTempPath();
         }
 
         [Fact]
         public async Task AugmentFiles_Test()
         {
-            File.Delete(Path.Combine(_scriptRootPath, "requirements.psd1"));
-            File.Delete(Path.Combine(_scriptRootPath, "profile.ps1"));
-            var powerShellFileProvisioner = new PowerShellFileProvisioner();
+            EnsurePowerShellFilesDoNotExist(_scriptRootPath);
+
+            var powerShellFileProvisioner = new PowerShellFileProvisionerMocksPSGalleryCalls(_logger);
+            await powerShellFileProvisioner.ProvisionFiles(_scriptRootPath);
+
+            File.Exists(Path.Combine(_scriptRootPath, RequirementsPsd1FileName));
+            File.Exists(Path.Combine(_scriptRootPath, ProfilePs1FileName));
+
+            string requirementsContent = File.ReadAllText(Path.Combine(_scriptRootPath, RequirementsPsd1FileName));
+
+            string expectedContent = @"# This file enables modules to be automatically managed by the Functions service.
```

Comment:
can you read contents for  src/WebJobs.Script/FileProvisioning/PowerShell/requirements.psd1 instead?

Reply:
Unfortunately, src/WebJobs.Script/FileProvisioning/PowerShell/requirements.psd1 does not have the final file content. I will move the string to constants.